vanmongo/collection.py

```diff
@@ -341,3 +341,12 @@ async def update_one(self, query: Dict[str, Any], update: Dict[str, Any] = {}):
     async def update_one_by_id(self, id: str, update: Dict[str, Any] = {}):
         """Update a document with specific ID"""
         return await self.update_one({"id": id}, update)
+
+    async def aggregate(
+        self, aggregation_pipeline: List[Dict[str, Any]]
+    ) -> AsyncGenerator[Any, None]:
+        """Perform aggregation on collection"""
+        cursor = self.collection.aggregate(aggregation_pipeline)
+
+        async for raw in cursor:
+            yield raw
```

Comment:
Maybe `aggregate` could return the pydantic model? With an `aggregate_raw` to return the raw dict?